### PR TITLE
VAR-184 | Always fetch reservations with start and end filters

### PR DESCRIPTION
**Pull request by badrange**
Before this change, if you go to manage reservations and change either start and end date, you'll get an empty page. The backend doesn't like queries with empty start or end dates. 

After this change, when you change start date (and end date is still empty), it automatically uses start date + 1 day as end date, and for end date, end date - 1 day for start date.

### CODE REVIEW

File: src/common/form/fields/DateField.js [hunk 1]
```diff
@@ -51,13 +52,20 @@ class UntranslatedDateField extends React.Component {
...
       locale,
       label,
       placeholder,
       value,
     } = this.props;
     const { isOpen } = this.state;
     const date = value || moment().toDate();
+    let valueOutput;
+
+    valueOutput = value ? moment(value).format('L') : null;
```
Comment (midori0507) [inline src/common/form/fields/DateField.js]: This is not a neat solution, this should move into a handler function instead of being as part of render

Comment (midori0507) [reply src/common/form/fields/DateField.js]: Dumb component actually need to be dumb, those calculation should be perform where this component get used.

TLDR: Those changes should not be placed here. This is a generic component

File: src/domain/reservation/manage/filters/ManageReservationsFilters.js [hunk 2]
```diff
@@ -40,6 +40,14 @@ class ManageReservationsFilters extends React.Component {
...
     } = this.props;
 
+    // Ensure that start and end are not empty. Otherwise we'll get an error from respa instead of search results.
+    if (filterName === 'start' && !filters.end) {
+      filters.end = filterValue;
+    }
+    if (filterName === 'end' && !filters.start) {
+      filters.start = filterValue;
+    }
+
```
Comment (midori0507) [inline src/domain/reservation/manage/filters/ManageReservationsFilters.js]: This check should be in DateField onChange handler, not here.
All filter changes, which mostly not related will have to go through those condition check

File: src/domain/reservation/manage/filters/ManageReservationsFilters.js [hunk 3]
```diff
@@ -48,6 +56,15 @@ class ManageReservationsFilters extends React.Component {
       newFilters[filterName] = filterValue;
     }
 
+    // We want to search for events that happen until the end of the day.
```
Comment (midori0507) [inline src/domain/reservation/manage/filters/ManageReservationsFilters.js]: Take this outside of this function

File: src/domain/reservation/manage/filters/ManageReservationsFilters.js [hunk 4]
```diff
@@ -136,6 +153,7 @@ class ManageReservationsFilters extends React.Component {
                 />
                 <div className="separator">-</div>
                 <DateField
+                  end
                   id="EndDateField"
                   label={t('ManageReservationsFilters.endDateLabel')}
                   onChange={(value) => {
```
Comment (midori0507) [inline src/domain/reservation/manage/filters/ManageReservationsFilters.js]: make an event handler here only for date selection change, put our logic to that
